x/swap/keeper/withdraw.go

```diff
@@ -35,12 +33,12 @@ func (k Keeper) Withdraw(ctx sdk.Context, owner sdk.AccAddress, shares sdk.Int,
 
 	poolRecord, found := k.GetPool(ctx, poolID)
 	if !found {
-		panic(fmt.Sprintf("pool %s not found", poolID))
+		return sdkerrors.Wrap(types.ErrPoolNotFound, poolID)
 	}
 
 	pool, err := types.NewDenominatedPoolWithExistingShares(poolRecord.Reserves(), poolRecord.TotalShares)
 	if err != nil {
-		panic(fmt.Sprintf("invalid pool %s: %s", poolID, err))
+		return sdkerrors.Wrapf(types.ErrInvalidPool, "%s: %s", poolID, err)
```

Comment:
Ditto here as well -- a stored pool must always be valid.  If it's not, it's a bug.